src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -0,0 +1,222 @@
+<script context="module" lang="ts">
+  declare global {
+    namespace JSX {
+      interface IntrinsicElements {
+        'leo-segmented-item': HTMLAttributes<HTMLElement> & {
+          // Note: This should line up with Reacts key type, but we don't want
+          // to depend on React in this layer, so we just define it manually.
+          key?: string | number | null
```

Comment:
I wonder if we should share this somewhere, as this is basically the third time we're defining this type.